v3/src/models/data/formula-fn-registry.ts

```diff
@@ -30,17 +30,19 @@ const cachedAggregateFnFactory =
 // The only difference is the final math operation applies to the expression results.
 const aggregateFnWithFilterFactory = (fn: (values: number[]) => number) => {
   return (args: MathNode[], mathjs: any, scope: FormulaMathJsScope) => {
-    const expression = args[0]
-      const filter = args[1]
-      let expressionValues = evaluateNode(expression, scope)
-      if (filter) {
-        const filterValues = evaluateNode(filter, scope)
-        expressionValues = expressionValues.filter((v: any, i: number) => !!filterValues[i])
-      }
-      return fn(expressionValues)
+    const [ expression, filter ] = args
+    let expressionValues = evaluateNode(expression, scope)
+    if (filter) {
+      const filterValues = evaluateNode(filter, scope)
+      expressionValues = expressionValues.filter((v: any, i: number) => !!filterValues[i])
+    }
+    return fn(expressionValues)
   }
 }
 
+// CODAP formulas assume that 0 is a truthy value, which is different from default JS behavior.
```

Comment:
One might add something like "so that, for instance, `count(attribute)` will return a count of valid data values, since 0 is a valid numeric value." to provide some context for the decision.

Reply:
Thanks, I've just added it to my next PR.